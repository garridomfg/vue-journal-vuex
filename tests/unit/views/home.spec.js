import { shallowMount } from '@vue/test-utils'
import Home from '@/views/Home'

describe('Home view test', () => {

    test('should render the component correctly', () => {
        const wrapper = shallowMount(Home)
        expect(wrapper.html()).toMatchSnapshot()
    });

    test('should redirect to no-entry on click', () => {
        const mockRouter = {
            push: jest.fn()
        }

        const wrapper = shallowMount(Home, {
            global: {
                mocks: {
                    $router: mockRouter
                }
            }
        })
        wrapper.find('button').trigger('click')
        expect(mockRouter.push).toHaveBeenCalled()
        expect(mockRouter.push).toHaveBeenCalledWith({name: 'no-entry'})
    });
});