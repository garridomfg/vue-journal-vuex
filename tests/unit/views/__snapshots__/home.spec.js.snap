// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home view test should render the component correctly 1`] = `
<div class="home"><img alt="Vue logo" src="">
  <h1>Hola mundo</h1><button class="btn btn-primary">Primary</button><button class="btn btn-secondary">Secondary</button><button class="btn btn-success">Succes</button>
</div>
`;
